### ISSUE TITLE
Ways to clean up C string literals before writing output?

### ISSUE THREAD

**Issue by anp** [immunant/c2rust#22]
This is pretty hard to parse even once you get used to it: `b"contents\x00" as *const u8 as *const c_char`. One way to reduce the noise while remaining correct would be to rely on a crate that lets you declare a C string literal like `cstr-macro`.

**Comment by rinon** [comment]
Going through and triaging old issues. This has been largely taken care of by refactoring passes that clean up the resulting rust code. I'll dig up some details and examples.

**Comment by ahomescu** [comment]
We could revisit this now that C string literals are in the language.

**Comment by Rua** [comment]
@ahomescu @kkysen I wanted to try to implement this, but as it stands, C-string literals are not even available as an unstable feature in the nightly currently used by c2rust. It would need a nightly from June 2023 at the earliest, preferably August.

**Comment by ahomescu** [comment]
I'm not against bumping the nightly if it's only for transpiler output (c2rust-refactor is stuck on 2022-08-08 and it seemed like a lot of work to move). Adding @thedataking for visibility.
